include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/Problems/IDEInstInteractionAnalysis.h

```diff
@@ -670,9 +734,56 @@ class IDEInstInteractionAnalysisT
   inline std::shared_ptr<EdgeFunction<l_t>>
   getCallToRetEdgeFunction(n_t callSite, d_t callNode, n_t retSite,
                            d_t retSiteNode, std::set<f_t> callees) override {
-    // Just forward to getNormalEdgeFunction() to check whether a user has
-    // additional labels for this call site.
-    return getNormalEdgeFunction(callSite, callNode, retSite, retSiteNode);
+    // Check if the user has registered a fact generator function
+    l_t UserEdgeFacts = BitVectorSet<e_t>();
+    std::set<e_t> EdgeFacts;
+    if (edgeFactGen) {
+      EdgeFacts = edgeFactGen(callSite);
+      // fill BitVectorSet
+      UserEdgeFacts = BitVectorSet<e_t>(EdgeFacts.begin(), EdgeFacts.end());
+    }
+    // Model call to heap allocating functions (new, new[], malloc, etc.) --
+    // only model direct calls, though.
+    if (callees.size() == 1) {
```

Comment:
What happens in the case we have non direct calls, maybe through an allocation? Not sure if this is correctly handled by the other code.

Reply:
Let's discuss that in our next meeting.